src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/service/PrisonerSearchApiService.kt

```diff
@@ -12,15 +12,26 @@ class PrisonerSearchApiService(
   private val prisonerSearchApiClient: PrisonerSearchApiClient,
 ) {
 
-  fun getAllPrisonersInPrison(prisonId: String): List<Prisoner> =
-    prisonerSearchApiClient.getPrisonersByPrisonId(
-      prisonId = prisonId,
-      page = PrisonerSearchApiClient.FIRST_PAGE,
-      pageSize = PrisonerSearchApiClient.DEFAULT_PAGE_SIZE,
-    ).content
+  fun getAllPrisonersInPrison(prisonId: String): List<Prisoner> {
+    var page = 0
+    val pageSize = 250
```

Comment:
I've opted for requests of 250 prisoners at a time  - there might be a more optimal number (or way of coming up with this number), but it seems reasonable enough for now and the use case at hand.